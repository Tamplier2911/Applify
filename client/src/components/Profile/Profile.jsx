// import "./Profile.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { selectUserObject } from "../../redux/auth/auth.selectors";

// components
import ProfilePortrait from "../ProfilePortrait/ProfilePortrait";
import ProfileAboutFrom from "../ProfileAboutForm/ProfileAboutForm";
import ProfileInfoForm from "../ProfileInfoForm/ProfileInfoForm";
import ProfilePasswordForm from "../ProfilePasswordForm/ProfilePasswordForm";

// JS Rendering CSS
import {
  ProfileContainer,
  ProfileTitle,
  ProfileContent
} from "./ProfileStyles";

// component constants
import profileData from "../../utils/ComponentProfileConstants/componentProfileConstants";

const Profile = ({ lang, user }) => {
  // console.log(user);
  const { name, email, photo, about } = user;
  const { profileTitle } = profileData[lang];

  // let image = "";
  // if (process.env.NODE_ENV === "development" && photo) {
  //   image = `${process.env.REACT_APP_SERVE_IMAGE_DEV + photo}`;
  // } else {
  //   image = photo;
  // }

  let image = "";
  if (process.env.NODE_ENV === "development" && photo) {
    image = `${process.env.REACT_APP_SERVE_IMAGE_DEV + "api/" + photo}`;
  } else {
    image = `${"api/" + photo}`;
  }

  console.log(image);
  return (
    <ProfileContainer>
      <ProfileTitle>{profileTitle}</ProfileTitle>
      <ProfileContent>
        <ProfilePortrait photo={image} />
        <ProfileAboutFrom about={about} />
        <ProfileInfoForm name={name} email={email} />
        <ProfilePasswordForm />
      </ProfileContent>
    </ProfileContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
  user: selectUserObject
});

export default connect(mapStateToProps)(Profile);
