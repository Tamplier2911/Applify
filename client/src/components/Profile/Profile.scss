/*

.profile {
  &__title {
    // ${getFont}
    font-family: var(--font-title);
    font-size: 4rem;
    font-weight: 300;
    padding: 1rem;

    @media only screen and (max-width: 768px) {
      line-height: 1;
    }
  }

  &__container {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-row-gap: 5rem;
    grid-column-gap: 5rem;

    @media only screen and (max-width: 768px) {
      grid-template-columns: 1fr;
      grid-row-gap: 2rem;
      grid-column-gap: 2rem;
    }

    &--imgWrap {
      justify-self: center;
      align-self: center;

      width: 20rem;
      height: 20rem;
      border-radius: 50%;
      box-shadow: 0 0.2rem 0.5rem #333;
      overflow: hidden;

      @media only screen and (max-width: 768px) {
        width: 14rem;
        height: 14rem;
        // border-radius: 0%;
      }
    }
    &--img {
      width: 100%;
      height: 100%;
      display: block;
      object-fit: cover;
    }
  }
}

*/
