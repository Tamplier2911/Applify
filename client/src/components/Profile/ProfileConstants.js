const profileData = {
  eng: {
    profileTitle: "Profile:",
  },
  rus: { profileTitle: "Профиль:" },
  ukr: { profileTitle: "Профіль:" },
};

export default profileData;
