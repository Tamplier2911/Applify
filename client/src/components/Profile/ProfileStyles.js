import styled, { css } from "styled-components";

const ProfileFadeIn = css`
  @keyframes profileFadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`;

const getFont = props => {
  const { lang } = props;
  if (lang === "eng") {
    return `font-family: var(--font-title);`;
  } else if (lang === "rus" || lang === "ukr") {
    return `font-family: var(--font-cyrillic);`;
  }
  return `font-family: var(--font-title);`;
};

export const ProfileContainer = styled.div`
  ${ProfileFadeIn}
  animation: 1s profileFadeIn
`;

export const ProfileTitle = styled.h2`
  ${getFont}
  font-size: 4rem;
  font-weight: 300;
  padding: 1rem;

  @media only screen and (max-width: 768px) {
    line-height: 1;
  }
`;

export const ProfileContent = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-row-gap: 5rem;
  grid-column-gap: 5rem;

  @media only screen and (max-width: 768px) {
    grid-template-columns: 1fr;
    grid-row-gap: 2rem;
    grid-column-gap: 2rem;
  }
`;
