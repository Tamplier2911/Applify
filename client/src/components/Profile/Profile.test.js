import React from "react";

// component
import Profile from "./Profile";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<Profile />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    user: {
      userObject: {
        id: "sdk897652",
        name: "Thomas White",
        email: "ex@mple.com",
        photo: "profilePhoto.jpg",
        about: "I love unicorns.",
        likedBlogposts: [],
      },
      errorMessage: null,
      isLogged: false,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<Profile store={store} />);
  });

  it("renders without an error", () => {
    const profile = findByTestAttr(wrapper, "profile");
    expect(profile.length).toBe(1);
  });

  it("renders - children profile content", () => {
    const profileContent = findByTestAttr(wrapper, "profile-content");
    expect(profileContent.length).toBe(1);
  });

  it("matches snapshot", () => {
    const profile = findByTestAttr(wrapper, "profile");
    expect(profile).toMatchSnapshot();
  });
});
