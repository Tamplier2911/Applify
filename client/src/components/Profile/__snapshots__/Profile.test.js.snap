// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<styled.div
  data-test="profile"
>
  <styled.h2>
    Profile:
  </styled.h2>
  <styled.div
    data-test="profile-content"
  >
    <ProfilePortrait
      photo="api/profilePhoto.jpg"
    />
    <Connect(ProfileAboutForm)
      about="I love unicorns."
    />
    <Connect(ProfileInfoForm)
      email="ex@mple.com"
      name="Thomas White"
      photo="profilePhoto.jpg"
    />
    <Connect(ProfilePasswordForm) />
  </styled.div>
</styled.div>
`;
